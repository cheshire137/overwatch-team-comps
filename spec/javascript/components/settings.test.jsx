import renderer from 'react-test-renderer'

import mockLocalStorage from '../mocks/local-storage'

import Settings from '../../../app/assets/javascripts/components/settings.jsx'

describe('Settings', () => {
  let component = null
  let store = null

  beforeEach(() => {
    store = {
      'overwatch-team-comps': JSON.stringify({
        email: 'user@example.com',
        region: 'eu',
        platform: 'xbl'
      })
    }
    mockLocalStorage(store)

    component = <Settings />
  })

  test('matches snapshot', () => {
    const tree = renderer.create(component).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
