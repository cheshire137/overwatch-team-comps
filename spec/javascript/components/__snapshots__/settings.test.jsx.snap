exports[`Settings matches snapshot 1`] = `
<div
  className="container">
  <form
    className="settings-form"
    onSubmit={[Function]}>
    <div
      className="field">
      <label
        className="label"
        htmlFor="email">
        Email address:
      </label>
      <input
        className="input"
        disabled={false}
        id="email"
        onChange={[Function]}
        placeholder="Your email address"
        type="text"
        value="user@example.com" />
    </div>
    <div
      className="field">
      <label
        className="label"
        htmlFor="platform">
        Which platform do you primarily play on?
      </label>
      <div>
        <span
          className="select ">
          <select
            disabled={false}
            id="platform"
            name="platform"
            onChange={[Function]}
            value="xbl">
            <option
              value="" />
            <option
              value="pc">
              PC
            </option>
            <option
              value="psn">
              PlayStation
            </option>
            <option
              value="xbl">
              Xbox
            </option>
          </select>
        </span>
      </div>
    </div>
    <div
      className="field">
      <label
        className="label"
        htmlFor="region">
        Which region do you primarily play in?
      </label>
      <div>
        <span
          className="select ">
          <select
            disabled={false}
            id="region"
            name="region"
            onChange={[Function]}
            value="eu">
            <option
              value="" />
            <option
              value="us">
              United States
            </option>
            <option
              value="eu">
              Europe
            </option>
            <option
              value="kr">
              South Korea
            </option>
            <option
              value="cn">
              China
            </option>
            <option
              value="global">
              Global
            </option>
          </select>
        </span>
      </div>
    </div>
    <button
      className="button is-primary"
      disabled={false}
      type="submit">
      Save
    </button>
  </form>
</div>
`;
